### PR TITLE
Add boomerang backups support

### PR DESCRIPTION
**Pull request by puzpuzpuz**
* Implements support of backup acks to client (a.k.a. boomerang backups)
* Adds related unit tests for `InvocationService` and `Invocation`
* Slightly improves code quality in `InvocationService`, `Invocation` and `ClientMessage`
* Also migrates `ClientLabelTest` to async/await syntax

Corresponding PRD: https://hazelcast.atlassian.net/wiki/spaces/PM/pages/1764032832/Node.js+Client+Boomerang+Backups

### Feature Overview

So far clients were waiting for the sync backups to complete on the member. This was causing 4 network hops to complete a client operation with sync backup. Since sync backup configuration is our out-of-box experience, we would like to improve it. Boomerang backup design decreases network hops to 3, thus improving the performance. 

boomerang-backups

### CODE REVIEW

File: src/invocation/InvocationService.ts [hunk 1]
```diff
@@ -237,49 +370,50 @@ export class InvocationService {
...
-
     /**
      * Removes the handler for all event handlers with a specific correlation id.
-     * @param id correlation id
      */
-    removeEventHandler(id: number): void {
-        if (this.eventHandlers.hasOwnProperty('' + id)) {
-            delete this.eventHandlers[id];
+    removeEventHandler(correlationId: number): void {
+        if (this.eventHandlers.has(correlationId)) {
```
Comment (mdumandag) [inline src/invocation/InvocationService.ts:377]: I think we can call `delete` without checking for the existence of `correlationId` as key

File: src/invocation/InvocationService.ts [hunk 2]
```diff
@@ -132,6 +163,73 @@ export class Invocation {
 
         return false;
     }
+
+    notify(clientMessage: ClientMessage): void {
+        assert(clientMessage != null, 'Response can not be null');
+        const expectedBackups = clientMessage.getNumberOfBackupAcks();
+        if (expectedBackups > this.backupsAcksReceived) {
+            this.pendingResponseReceivedMillis = Date.now();
```
Comment (mdumandag) [inline src/invocation/InvocationService.ts:171]: This block is executed once per the invocation, right?

**Review by mdumandag — COMMENTED** [review]
Looks good to me. I have a question and a minor comment

File: src/invocation/InvocationService.ts (hunk 1 above)
Comment (puzpuzpuz) [reply src/invocation/InvocationService.ts]: Good point. Fixed in 615278a59d65da7672c6120d9b7156e0f2726259

File: src/invocation/InvocationService.ts (hunk 2 above)
Comment (puzpuzpuz) [reply src/invocation/InvocationService.ts:171]: Yes, it's executed once per response from the target member.

**Review by mdumandag — APPROVED** [review]

**Pull request merged**